networkx/algorithms/covering.py

```diff
@@ -0,0 +1,131 @@
+#    Copyright 2016 NetworkX developers.
+#    Copyright (C) 2004-2016 by
+#    Nishant Nikhil <nishantiam@gmail.com>
+#    All rights reserved.
+#    BSD license.
+
+""" Functions related to graph covers."""
+
+import networkx as nx
+from networkx.utils import not_implemented_for, arbitrary_element
+
+__all__ = ['min_edge_cover', 'is_edge_cover', 'is_min_edge_cover']
+
+
+@not_implemented_for('directed')
+@not_implemented_for('multigraph')
+def min_edge_cover(G, matching_algorithm=None):
+    r"""Returns a set of edges which constitutes
+    the minimum edge cover of the graph.
+
+    A smallest edge cover can be found in polynomial time by finding
+    a maximum matching and extending it greedily so that all vertices
+    are covered.
+
+    Parameters
+    ----------
+    G : An undirected, unweighted graph.
+
+    Returns
+    -------
+    min_cover : set
+
+        The covering is returned as a set, ``min_cover``.
+        It contains all the edges of minimum edge cover
+        in form of tuples.
+
+    matching_algorithm : function
+        It specifies the maximum weight matching algorithm to be used.
+
+    Notes
+    -----
+    An edge cover of a graph is a set of edges such that every vertex of
+    the graph is incident to at least one edge of the set.
+    A minimum edge cover is an edge covering of smallest possible size.
+
+    Due to its implementation, the worst-case running time of this algorithm
+    is bounded by the worst-case running time of the function
+    `matching_algorithm`.
+
+    Minimum edge cover for bipartite graph can also be found using the
+    function present in :mod:`networkx.algorithms.bipartite.covering`
+    """
+    if nx.number_of_isolates(G) > 0:
+        # ``min_cover`` does not exist as there is an isolated vertex
+        raise nx.NetworkXException(
+            "Graph has a vertex with no edge incident on it, "
+            "so no edge cover exists.")
+    if matching_algorithm is None:
+        matching_algorithm = nx.max_weight_matching
+    maximum_matching = matching_algorithm(G)
+    # ``min_cover`` is superset of ``maximum_matching``
+    min_cover = set(maximum_matching.items())
+    # iterate for uncovered nodes
+    uncovered_nodes = set(G) - set(v for u, v in min_cover)
```

Comment:
You can use a set comprehension instead, it should be a bit faster: `{v for u, v in min_cover}`.